lib/mongoid/fields/standard.rb

```diff
@@ -231,7 +231,11 @@ def included?(fields)
       # @since 2.4.4
       def evaluated_default(doc)
         if default_val.respond_to?(:call)
-          evaluate_default_proc(doc)
+          begin
+            evaluate_default_proc(doc)
+          rescue NoMethodError => e
+            return nil
+          end
```

Comment:
Thanks for the pull request! 
I think instead of catching a specific error here, we would want to do a more general change so that we could solve the problem noted here as well: https://github.com/mongodb/mongoid/blob/master/lib/mongoid/document.rb#L113-L114